### ISSUE TITLE
format() is failing to replace the replacement fields if the value is None

### ISSUE THREAD

**Issue by CarstenGrohmann** [QQuick/Transcrypt#515]
Hi,

Unlike Python, Transcrypt cannot `format()` the value `None` to a string. Other values like empty lists, `0` or empty dictionaries will be formatted like Python.

**Testlet output:**
```
!!myproj/testlet_format.py:2	'a: None; b: 1'	'a: {}; b: 1'	autotest.js:2530
!!!myproj/testlet_format.py:3	'a: 1; b: None'	'a: 1; b: {}'	autotest.js:2530
!!!myproj/testlet_format.py:4	'a: 1; b: None'	'a: 1; b: {1}'	autotest.js:2530
myproj/testlet_format.py:5	'a: 1; b: []'	'a: 1; b: []'	autotest.js:2530
myproj/testlet_format.py:6	'a: 1; b: []'	'a: 1; b: []'	autotest.js:2530
myproj/testlet_format.py:7	'a: 1; b: {}'	'a: 1; b: {}'	autotest.js:2530
myproj/testlet_format.py:8	'a: 1; b: {}'	'a: 1; b: {}'	autotest.js:2530
myproj/testlet_format.py:9	'a: 1; b: 0'	'a: 1; b: 0'	autotest.js:2530
myproj/testlet_format.py:10	'a: 1; b: 0'	'a: 1; b: 0'	autotest.js:2530
```

**Use this testlet to reproduce the issue:**
```
def run(autoTester):
    autoTester.check('a: {}; b: {}'.format(None, 1))
    autoTester.check('a: {}; b: {}'.format(1, None))
    autoTester.check('a: {0}; b: {1}'.format(1, None))
    autoTester.check('a: {0}; b: {1}'.format(1, []))
    autoTester.check('a: {}; b: {}'.format(1, []))
    autoTester.check('a: {0}; b: {1}'.format(1, {}))
    autoTester.check('a: {}; b: {}'.format(1, {}))
    autoTester.check('a: {0}; b: {1}'.format(1, 0))
    autoTester.check('a: {}; b: {}'.format(1, 0))
```

Regards,
Carsten

**Comment by JdeH** [comment]
Thanks, I'll correct it!

**Comment by neriusmika** [comment]
Jacques, look at #494

**Comment by JdeH** [comment]
Oops, that one slipped through...

**Comment by JdeH** [comment]
Fixed in 3.7.12
